libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -99,7 +97,7 @@ string IRGenerator::generate(ContractDefinition const& _contract)
 
 	t("CreationObject", creationObjectName(_contract));
 	t("memoryInit", memoryInit());
-	t("constructor", constructorCode(_contract));
+	t("constructor", _contract.isLibrary() ? "" : constructorCode(_contract));
 	t("deploy", deployCode(_contract));
 	generateQueuedFunctions();
 	t("functions", m_context.functionCollector().requestedFunctions());
```

Comment:
Is skipping the constructor enough for libraries or does this require some other changes?

Reply:
There are some further subtleties like the call protection, but for now it should be fine.